### PR TITLE
fixed sampler bank file dialog export

### PR DESCRIPTION
**Pull request by midzer**
https://bugs.launchpad.net/mixxx/+bug/1403223

Although i did not find the dialog in question, i realized it is the same issue with QFileDialog under Linux as https://github.com/mixxxdj/mixxx/pull/430

I am unsure whether bug occurs under same environment in void SamplerBank::slotLoadSamplerBank() as well. Might need further attention.

### CODE REVIEW

**Comment by midzer** [comment]
just added the comment as suggested by @kain88-de

File: src/samplerbank.cpp [hunk 1]
```diff
@@ -29,14 +29,24 @@ void SamplerBank::slotSaveSamplerBank(double v) {
...
-
+    QString filefilter = tr("Mixxx Sampler Banks (*.xml)");
     QString samplerBankPath = QFileDialog::getSaveFileName(
             NULL, tr("Save Sampler Bank"),
             QString(),
-            tr("Mixxx Sampler Banks (*.xml)"));
-    if (samplerBankPath.isEmpty()) {
+            tr("Mixxx Sampler Banks (*.xml)"),
+            &filefilter);
+    if (samplerBankPath.isNull() || samplerBankPath.isEmpty()) {
```
Comment (kain88-de) [inline src/samplerbank.cpp]: It is enough to check for `isEmpty` with a string.

File: src/samplerbank.cpp [hunk 2]
```diff
@@ -29,14 +29,24 @@ void SamplerBank::slotSaveSamplerBank(double v) {
...
     }
+    // Manually add extension due to bug in QFileDialog
+    // via https://bugreports.qt-project.org/browse/QTBUG-27186
+    // Can be removed after switch to Qt5
+    QFileInfo fileName(samplerBankPath);
+    if (fileName.suffix().isNull() || fileName.suffix().isEmpty()) {
+    	QString ext = filefilter.section(".",1,1);
+        ext.chop(1);
+        samplerBankPath.append(".").append(ext);
+        }
```
Comment (kain88-de) [inline src/samplerbank.cpp]: Please check the indentation of this section.

**Comment by midzer** [comment]
I removed unnecessary check and hope intendation is fixed now.

**Comment by midzer** [comment]
@kain88-de I removed unnecessary check and hope intendation is fixed now.

File: src/samplerbank.cpp [hunk 3]
```diff
@@ -29,14 +29,24 @@ void SamplerBank::slotSaveSamplerBank(double v) {
...
     }
+    // Manually add extension due to bug in QFileDialog
+    // via https://bugreports.qt-project.org/browse/QTBUG-27186
+    // Can be removed after switch to Qt5
+    QFileInfo fileName(samplerBankPath);
+    if (fileName.suffix().isEmpty()) {
+    	QString ext = filefilter.section(".",1,1);
+    	ext.chop(1);
+        samplerBankPath.append(".").append(ext);
+        }
```
Comment (kain88-de) [inline src/samplerbank.cpp]: **extrem nitpicking mode**

line 45 and 46 contain tabs and the closing `}` is not intened correctly. Could you change that as well.

I'm not sure which editor you use. But assuming vim and you have defined tabs as 4 space characters you can use 'retab' to remove the tabs. To fix the indentation in an automated way use mark all lines of the if statement and hit `=`.

**Comment by midzer** [comment]
@kain88-de next try. i disabled tab intendation in Eclipse now.

**Comment by kain88-de** [comment]
Yes it looks good now. Thanks

**Pull request merged**